snoop/data/analyzers/archives.py

```diff
@@ -196,20 +194,28 @@ def unarchive(blob):
 def archive_walk(path):
     """Generates simple dicts with archive listing for the archive. """
 
-    for thing in path.iterdir():
-        if thing.is_dir():
+    for entry in os.scandir(path):
+        if entry.is_dir(follow_symlinks=False):
             yield {
                 'type': 'directory',
-                'name': thing.name,
-                'children': sorted(
-                    list(archive_walk(thing)),
-                    key=lambda c: c['name'],
-                ),
+                'name': entry.name,
+                'children': list(archive_walk(entry.path)),
             }
-
         else:
             yield {
                 'type': 'file',
-                'name': thing.name,
-                'blob_pk': models.Blob.create_from_file(thing).pk,
+                'name': entry.name,
+                'path': entry.path
             }
+
+
+def create_blobs(dirlisting):
+    """Create blobs for files in archive listing created by archive_walk"""
```

Comment:
Add links like this: `[archive_walk][snoop.data.analyzers.archive_walk]`.

End docstring with a `.` because that's what the docstring style looks like.